keras/preprocessing/text.py

```diff
@@ -45,11 +46,59 @@ def one_hot(text, n,
             filters='!"#$%&()*+,-./:;<=>?@[\\]^_`{|}~\t\n',
             lower=True,
             split=' '):
+    """One-hot encode a text into a list of word indexes in a vocabulary of
+    size n (unicity of word to index mapping non-guaranteed).
+
+    This is a wrapper to the `hashing_trick` function using `hash` as the
+    hashing function.
+    """
+    return hashing_trick(text, n,
+                         hash_function=hash,
+                         filters=filters,
+                         lower=lower,
+                         split=split)
+
+
+def hashing_trick(text, n,
+                  hash_function=None,
+                  filters='!"#$%&()*+,-./:;<=>?@[\\]^_`{|}~\t\n',
+                  lower=True,
+                  split=' '):
+    """Converts a text to a sequence of indices in a fixed-size hashing space
+
+    # Arguments
+        text: Input text (string).
+        n: Dimension of the hashing space.
+        hash_function: The hash function to use. Takes in input a string,
```

Comment:
Should this argument accepted a string from a predefined set, such as `"md5"`? What would be some good (fast) functions here?